### PR TITLE
Support compression= in DataFrame.to_json

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
closes https://github.com/rapidsai/cudf/issues/17564

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/io/json.py [hunk 1]
```diff
@@ -54,6 +54,22 @@ def _get_cudf_schema_element_from_dtype(
     return lib_type, child_types
 
 
+def _to_plc_compression(
```
Comment (Matt711) [inline python/cudf/cudf/io/json.py:57]: Do we use something like this in other places? If so, we could reuse it.

File: python/cudf/cudf/tests/test_json.py [hunk 2]
```diff
@@ -1453,3 +1453,12 @@ def test_chunked_json_reader():
     with cudf.option_context("io.json.low_memory", True):
         gdf = cudf.read_json(buf, lines=True)
     assert_eq(df, gdf)
+
+
+@pytest.mark.parametrize("compression", ["gzip", None])
```
Comment (Matt711) [inline python/cudf/cudf/tests/test_json.py:1458]: There's a `compression_params` you could use

```suggestion
@pytest.mark.parametrize("compression", compression_params)
```

**Review by Matt711 — APPROVED** [review]

File: python/cudf/cudf/io/json.py (hunk 1 above)
Comment (mroeschke) [reply python/cudf/cudf/io/json.py:57]: It looks like we have something similar we use in `parquet`, but given that each format supports different compression (and maps slightly differently from Python), I can take a look at this in a follow up

**Pull request merged**